sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,532 @@
+                    preds.append(output.forecast.detach().cpu().numpy())
+                    histories.append(timeseries.detach().cpu().numpy())
+
+            # losses = np.array(losses)
+            # average_loss = np.average(losses)
+            self._model.train()
+
+            trues = np.concatenate(trues, axis=0)
+            preds = np.concatenate(preds, axis=0)
+            histories = np.concatenate(histories, axis=0)
+
+            # metrics = get_forecasting_metrics(y=trues, y_hat=preds, reduction="mean")
+
+            # print(
+            #     f"Epoch {cur_epoch}: Test MSE: {metrics.mse:.3f}",
+            #     f"|Test MAE: {metrics.mae:.3f}",
+            # )
+        return self
+
+    def _predict(self, y, fh=1, X=None):
+        """Predict method to forecast timesteps into the future.
+
+        fh must be the same length as the one used to fit the model.
+        """
+        from torch import from_numpy
+
+        self._model = self._model.to(self.device)
+        self._model.eval()
+        # first convert it into numpy values
+        y_ = y.values
+        sequence_length, num_channels = y_.shape  # shape of our input to predict
+        # raise warning if sequence length of y is greater than the sequence
+        # length used to fit the model
+        if sequence_length > self._seq_len:
+            warnings.warn(
+                f"length of {y.shape[0]} was found which is greater than sequence "
+                "length {self._seq_len} used to fit the model. The most recent"
+                f" {self._seq_len} values will be used.",
+                stacklevel=2,
+            )
+            # only retain the most recent self._seq_len values if greater than
+            # self._seq_len
+            y_ = y_[-self._seq_len :, :]
+            sequence_length = self._seq_len
+
+        # transpose it to change it into (C, S) size
+        y_ = y_.T
+        if sequence_length < self._moment_seq_len:
+            # if smaller, need to pad values
+            y_ = _create_padding(
+                y_, (self._moment_seq_len - sequence_length, num_channels)
+            )
+            input_mask = _create_mask(
+                sequence_length, self._moment_seq_len - sequence_length
+            )
+        else:  # this means sequence_length = self._seq_len == 512
+            input_mask = _create_mask(self._moment_seq_len)
+        if num_channels != self._y_shape[1]:
+            # Todo raise error here
+            pass
+        # returns a timeseriesoutput object
+        y_torch_input = (
+            from_numpy(y_.reshape((1, self._y_shape[1], -1))).float().to(self.device)
+        )
+        input_mask = input_mask.to(self.device)
+        output = self._model(y_torch_input, input_mask)
+        forecast_output = output.forecast
+        forecast_output = forecast_output.squeeze(0)
+
+        pred = forecast_output.detach().cpu().numpy().T
+
+        df_pred = pd.DataFrame(pred, columns=self._y_cols)
```

Comment:
you need to set the index of the data

Reply:
isnt the index the length of the fh? so it ranges from 0-(fh-1)